### PR TITLE
Explore: add URL schema docs

### LINKED ISSUE #70432: Explore: add documentation for query parameters and URL schema
With https://github.com/grafana/grafana/pull/69692 we introduced strong versioning for URL schemas. We should document explicitly and define a contract on what's expected to be in the URL and how to generate them in case external tools rely on that logic.

### PR DESCRIPTION
**Pull request by Elfo404**
**What is this feature?**

Adds brief documentation about Explore URL schema. 
At a later stage we will provide an actual schema in `@grafana/schema` for those types, but requires some refactoring in types in Explore

**Why do we need this feature?**

The URL schema in Explore is not documented, this will hopefully avoid situations in which users generate URLs that are not fully compliant with what Explore expects.

**Which issue(s) does this PR fix?**:

Fixes #70432 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/explore/_index.md
Comment (gelicia) [inline docs/sources/explore/_index.md:101]: Would we want to add a comment here about schema version 0 being for compact URLs? 

"Some older Explore URLs may be in "compact" format, which look like [...]. While we do support these URLs, we migrate the URLs to be in schema version 1. We recommend replacing the source of compact format URLs to match the format defined above."

File: docs/sources/explore/_index.md
Comment (gelicia) [inline docs/sources/explore/_index.md:77]: If we go with mentioning compact URLs below, we should clarify this a little. 

"`schema_version` is the schema version. Currently, we migrate URLs to version `1`"

File: docs/sources/explore/_index.md
Comment (Elfo404) [reply docs/sources/explore/_index.md:101]: we discussed this briefly yesterday during standup, the idea for this doc is to only describe the current version as a guide for who wants to build a valid Explore URL. older schemas may better live in types definitions for when we will actually use the schema generator and push those types in `@grafana/schema`, or in older versions of the docs

File: docs/sources/explore/_index.md
Comment (Elfo404) [reply docs/sources/explore/_index.md:77]: similarly to https://github.com/grafana/grafana/pull/71724#discussion_r1267924296, i think the objective here is to describe how to generate one for the current version without providing too many details about the whys.

probably however ``currently only `1` is supported`` is a bit confusing as it implies that there will be other schema versions with the same structure. maybe we should rephrase this to something like:

```suggestion
- `schema_version` is the schema version of the URL format, to make Explore parse the URL using the current schema it must be set to `1`
```

maybe? wdyt? (cc. @ifrost @harisrozajac )

Comment (harisrozajac) [reply docs/sources/explore/_index.md:77]: > - `schema_version` is the schema version of the URL format, to make Explore parse the URL using the current schema it must be set to `1` 

@Elfo404 I think this is a good description. Since it implies that there are other schema versions, would it make sense to specify somewhere what those are?

File: docs/sources/explore/_index.md
Comment (ifrost) [inline docs/sources/explore/_index.md:68]:
```suggestion
Because Explore URLs have a defined structure, you can build a URL from external tools and open it in Grafana. The URL structure is:
```

File: docs/sources/explore/_index.md
Comment (ifrost) [reply docs/sources/explore/_index.md:77]: Maybe?

```suggestion
- `schema_version` is the schema version (should be set to the latest version which is `1`)
```

File: docs/sources/explore/_index.md
Comment (ifrost) [inline docs/sources/explore/_index.md:86]: Is it actually required when a non-mixed data source is used at the root? If not maybe we can mention it, otherwise it might be confusing why it needs to be set up twice when non-mixed is used.

File: docs/sources/explore/_index.md
Comment (ifrost) [inline docs/sources/explore/_index.md:92]: I guess `queries` value escaped from that array!

File: docs/sources/explore/_index.md
Comment (ifrost) [reply docs/sources/explore/_index.md:101]: I think it makes sense to have official documentation only for the latest. We support older versions because of backward compatibility, but they are not officially supported as alternatives.

File: docs/sources/explore/_index.md
Comment (Elfo404) [reply docs/sources/explore/_index.md]: this is how prettier formats that `[]` after the object if defined as typescript. 
i removed the file type and updated the block in 7998e094af, should be a bit better now

File: docs/sources/explore/_index.md
Comment (Elfo404) [reply docs/sources/explore/_index.md:90]: not strictly required as Explore will use the current root datasource if the query doen't have the datasource property, but that was done for legacy reasons. when not defining a datasource in each query explore will update the URL to include on for each, so effectively the URL will match this schema, not one without that property

Comment (ifrost) [reply docs/sources/explore/_index.md:90]: Then maybe we could add a bit more context for both, e.g.:

```typescript
datasource: string; // Explore's root datasource UID, or `-- Mixed --` for mixed datasources
...
uid: string; // query's datasource UID ie: "AD7864H6422"
```

Comment (Elfo404) [reply docs/sources/explore/_index.md:90]: updated in d05619da00

**Review by ifrost — APPROVED** [review]

File: docs/sources/explore/_index.md
Comment (knylander-grafana) [inline docs/sources/explore/_index.md:103]: I'm updating the note to use the admonition shortcode. We changed the Note format to be an admonition shortcode (see [Shortcodes](https://grafana.com/docs/writers-toolkit/write/shortcodes/#admonition-shortcode) content in the Writers' Toolkit). 
```suggestion
{{% admonition type="note" %}}
The `from` and `to` also accept relative ranges defined in [Time units and relative ranges]({{< relref "../dashboards/use-dashboards/#time-units-and-relative-ranges" >}}).
{{% /admonition %}}
```

Comment (Elfo404) [reply docs/sources/explore/_index.md]: thanks!

File: docs/sources/explore/_index.md
Comment (knylander-grafana) [inline docs/sources/explore/_index.md:69]: This suggestion moves the phrasing towards active 
```suggestion
When using Explore, the URL in the browser address bar updates as you make changes to the queries. You can share or bookmark this URL. 
```

**Review by knylander-grafana — APPROVED** [review]
Thank you for adding doc! I am approving the doc. I'll let someone with merge rights also review the PR so you'll be able to merge these changes when you're ready.

**Review by Eve832 — APPROVED** [review]

**Pull request merged**
